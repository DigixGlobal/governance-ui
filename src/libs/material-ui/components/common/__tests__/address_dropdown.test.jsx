import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddressDropdown from '../address_dropdown';

jest.mock('react-redux', () => ({
  connect: () => component => component,
}));

const addresses = [{
  name: 'address 1',
  address: 'address test 1',
}, {
  name: 'address 2',
  address: 'address test 2',
}, {
  name: 'address 3',
  address: 'address test 4',
}];

describe('<AddressDropdown />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <AddressDropdown
        addresses={addresses}
        value="none"
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  test('renders correctly with matching value', () => {
    const component = shallow(
      <AddressDropdown
        addresses={addresses}
        value="address test 4"
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
