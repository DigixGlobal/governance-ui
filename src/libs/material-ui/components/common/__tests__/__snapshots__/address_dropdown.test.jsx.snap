// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressDropdown /> renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  floating={true}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  onChange={null}
  openOnFocus={true}
  options={
    Array [
      Object {
        "content": <div>
          address 1
        </div>,
        "key": "address test 1",
        "text": "address 1",
        "value": "address test 1",
      },
      Object {
        "content": <div>
          address 2
        </div>,
        "key": "address test 2",
        "text": "address 2",
        "value": "address test 2",
      },
      Object {
        "content": <div>
          address 3
        </div>,
        "key": "address test 4",
        "text": "address 3",
        "value": "address test 4",
      },
    ]
  }
  placeholder="Select Address"
  renderLabel={[Function]}
  search={true}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={true}
  tabIndex="-1"
/>
`;

exports[`<AddressDropdown /> renders correctly with matching value 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  floating={true}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  onChange={null}
  openOnFocus={true}
  options={
    Array [
      Object {
        "content": <div>
          address 1
        </div>,
        "key": "address test 1",
        "text": "address 1",
        "value": "address test 1",
      },
      Object {
        "content": <div>
          address 2
        </div>,
        "key": "address test 2",
        "text": "address 2",
        "value": "address test 2",
      },
      Object {
        "content": <div>
          address 3
        </div>,
        "key": "address test 4",
        "text": "address 3",
        "value": "address test 4",
      },
    ]
  }
  placeholder="Select Address"
  renderLabel={[Function]}
  search={true}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={true}
  tabIndex="-1"
  value="address test 4"
/>
`;
